volatility/framework/automagic/linux.py

```diff
@@ -293,6 +294,56 @@ def virtual_to_physical_address(cls, addr: int) -> int:
             return addr - 0xffffffff80000000
         return addr - 0xc0000000
 
+    @classmethod
+    def mask_mods_list(cls, context: interfaces.context.ContextInterface, layer_name: str,
+                       mods: Iterator[Any]) -> Iterator[Any]:
+        """
+        A helper function to mask the starting and end address of kernel modules
+        """
+        mask = context.layers[layer_name].address_mask
+
+        return [(utility.array_to_string(mod.name), mod.get_module_base() & mask, (mod.get_module_base() & mask) + mod.get_core_size())
+                for mod in mods]
+
+    @classmethod
+    def generate_kernel_handler_info(
+            cls,
+            context: interfaces.context.ContextInterface,
+            layer_name: str,
+            kernel,  # ikelos - how to type this??
```

Comment:
Ah, but that would then put the dependency for lsmod into the automagic.  5:S  Hmmmm... yeah, ok, let's go with the interface definitions then.  I think it would be possible to put the requirement into the automagic, but I've no idea how well errors would be reported that way...

Reply:
Hmmm, turns out looks like code like this already present on the mac side of things (including the comment about typing).  Might be good to fix both of these at once?

Reply:
Yes, this is bringing over the same functionality to Linux as the Mac plugins currently have/use.

What do you propose for a final fix?

Reply:
So my preference would be to have both take in the context and then only simple parameters (layer names, symbol names, etc), and then the code turn those into kernels/lists/etc.  I realize that may not be feasible with `mods_list`, since it might be very useful to have that as a changable parameter, but kernel feels like it could definitely be a string rather than an object?  I feel like I would've responded to this on the mac one if my name were in there, but regardless, it'd be good to get them both fixed up.

Reply:
Ok so if kernel was a string, then this code (same API as we are talking about for Linux, so should be the same for discussion purposes):

https://github.com/volatilityfoundation/volatility3/blob/master/volatility/framework/plugins/mac/check_syscall.py#L36

instead of taking the kernel object, it would need to get config config['darwin'] (already has config['primary']), and then call contexts.Module on it again like check_syscall does on line 32? check_syscalls needs kernel too, so we couldn't just have one call.

Reply:
Yes, in this one instance it would mean constructing the kernel object twice, because you'd be handing in a string at the first call, and then later on the generator requires specific symbols which it gets from the kernel.  That shouldn't be heavy on either memory or speed, but it would mean that the method is consistent with most every other method we've got, which gives coders less to have to remember.

I've just done a similar thing recently in #269, and whilst it's not technically as lean and efficient as it can be, I think the code reuse and sharing benefits outweigh the minor one off cost of constructing the kernel twice.

As a way of reducing the burden, it might be worth adding a method (as in #269) to do the kernel module construction because it's quite a common task?  The question will be where to put it...